crates/matrix-sdk-crypto/src/olm/account.rs

```diff
@@ -835,6 +840,43 @@ impl Account {
         }
     }
 
+    #[instrument(
+        skip_all,
+        fields(
+            user_id = %device.user_id(),
+            device_id = %device.device_id(),
+            algorithms = ?device.algorithms()
+        )
+    )]
+    fn find_pre_key_bundle(
+        device: &ReadOnlyDevice,
+        key_map: &BTreeMap<OwnedDeviceKeyId, Raw<ruma::encryption::OneTimeKey>>,
+    ) -> Result<PrekeyBundle, SessionCreationError> {
+        let mut keys = key_map.iter();
+
+        let first_key = keys.next().ok_or_else(|| {
+            SessionCreationError::OneTimeKeyMissing(
+                device.user_id().to_owned(),
+                device.device_id().into(),
+            )
+        })?;
+
+        let first_key_id = first_key.0.to_owned();
+        let first_key = OneTimeKey::deserialize(first_key_id.algorithm(), first_key.1)?;
+
+        let ret = match first_key {
+            OneTimeKey::SignedKey(key) => Ok(PrekeyBundle::Olm3DH { key }),
+            _ => Err(SessionCreationError::OneTimeKeyUnknown(
+                device.user_id().to_owned(),
+                device.device_id().into(),
+            )),
+        };
+
+        trace!(result = ?ret, "Finished searching for a valid pre-key bundle");
+
+        ret
```

Comment:
```suggestion
        let result = match first_key {
            OneTimeKey::SignedKey(key) => Ok(PrekeyBundle::Olm3DH { key }),
            _ => Err(SessionCreationError::OneTimeKeyUnknown(
                device.user_id().to_owned(),
                device.device_id().into(),
            )),
        };

        trace!(?result, "Finished searching for a valid pre-key bundle");

        result
```

Reply:
Could also use `ret(level = Level::TRACE)` in `#[instrument]` but then we'd lose the custom message.